### PR TITLE
try adding minimal IntelliJ project config

### PR DESCRIPTION
**Pull request by gavinking**
This only includes:

- coding conventions, and
- configuration of warnings.

[Please describe here what your change is about]

----------------------
By submitting this pull request, I confirm that my contribution is made under the terms of the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt)
and can be relicensed under the terms of the [LGPL v2.1 license](https://www.gnu.org/licenses/old-licenses/lgpl-2.1.txt) in the future at the maintainers' discretion.
For more information on licensing, please check [here](https://github.com/hibernate/hibernate-orm/blob/main/CONTRIBUTING.md#legal).

----------------------

### CODE REVIEW

**Comment by gavinking** [comment]
I believe you can have multiple of them and select the one you want.

**Comment by sebersole** [comment]
We could include `.idea/copyright` as well.  Yes spotless ultimately handles this, but it also handles things being configured here in `.idea/codeStyles`

**Comment by sebersole** [comment]
> I believe you can have multiple of them and select the one you want.

That's fair, but "Project" happens to be the default IJ chooses for a project-level Settings profile.  Maybe let's name this "Hibernate"?  I'd also suggest a few more, I left as a line comment

File: .idea/inspectionProfiles/Project_Default.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<component name="InspectionProjectProfileManager">
+  <profile version="1.0">
+    <option name="myName" value="Project Default" />
+    <inspection_tool class="BooleanMethodIsAlwaysInverted" enabled="false" level="WARNING" enabled_by_default="false" />
+    <inspection_tool class="DeprecatedIsStillUsed" enabled="true" level="WEAK WARNING" enabled_by_default="true" editorAttributes="INFO_ATTRIBUTES" />
+    <inspection_tool class="MarkedForRemoval" enabled="true" level="WEAK WARNING" enabled_by_default="true" editorAttributes="INFO_ATTRIBUTES" />
+    <inspection_tool class="JUnitMalformedDeclaration" enabled="false" level="ERROR" enabled_by_default="false" />
+    <inspection_tool class="StringBufferReplaceableByString" enabled="false" level="WARNING" enabled_by_default="false" />
+  </profile>
+</component>
```
Comment (sebersole) [inline .idea/inspectionProfiles/Project_Default.xml:10]:
```suggestion
    <inspection_tool class="JavadocBlankLines" enabled="true" level="WEAK WARNING" enabled_by_default="true" editorAttributes="INFO_ATTRIBUTES" />
    <inspection_tool class="JavadocDeclaration" enabled="true" level="WARNING" enabled_by_default="true">
      <option name="ADDITIONAL_TAGS" value="todo,settingDefault,remove" />
    </inspection_tool>
</component>

```

**Comment by lucamolteni** [comment]
> This only includes:
> 
> * coding conventions, and
> * configuration of warnings.

I've imported this in IDEA https://github.com/hibernate/hibernate-ide-codestyles/blob/main/intellij-14/hibernate_orm.xml

even though it's quite ancient it seems to be working

**Comment by gavinking** [comment]
> even though it's quite ancient it seems to be working

I had a look, but it seems to have quite a lot of unnecessary cruft: for example, all sorts of formatting rules for Groovy.

**Comment by gavinking** [comment]
> We could include `.idea/copyright` as well. 

Wow, this  amazingly complicated to configure in IntelliJ. Better if you do it.

> Maybe let's name this "Hibernate"? 

Tried this but unfortunately it doesn't work. It seems that you're only allowed to give names to profiles stored in the IDE, not to the "project" profile. But that's OK, I think. You can still store as many profiles as you like in the IDE, and even make one based off of the project one.

**Comment by gavinking** [comment]
Going to go ahead and merge what I have here so you guys can try it out and let me know if anything breaks. It seems to work nicely when I create a new project from a clean checkout.

**Comment by lucamolteni** [comment]
> > even though it's quite ancient it seems to be working
> 
> I had a look, but it seems to have quite a lot of unnecessary cruft: for example, all sorts of formatting rules for Groovy.

From a user perspective having two styles in two different places is confusing, perhaps you should remove the other one then?

**Pull request merged**